build-support/bin/native/bootstrap.sh

```diff
@@ -95,17 +103,21 @@ function bootstrap_native_code() {
   # Bootstraps the native code and overwrites the native_engine_version to the resulting hash
   # version if needed.
   local native_engine_version="$(calculate_current_hash)"
```

Comment:
and regarding tests - this is pretty well covered as a fundamental dependency of basically all v2 engine integration tests (all the tests break when this is broken). but would be happy to add tests if there's a specific area you think would benefit from it - maybe we can discuss more offline.